crates/uv-workspace/src/pyproject.rs

```diff
@@ -70,18 +70,43 @@ impl PyProjectToml {
     /// non-package ("virtual") project.
     pub fn is_package(&self) -> bool {
         // If `tool.uv.package` is set, defer to that explicit setting.
-        if let Some(is_package) = self
-            .tool
-            .as_ref()
-            .and_then(|tool| tool.uv.as_ref())
-            .and_then(|uv| uv.package)
-        {
+        if let Some(is_package) = self.is_uv_package() {
             return is_package;
         }
 
         // Otherwise, a project is assumed to be a package if `build-system` is present.
         self.build_system.is_some()
     }
+
+    /// Returns whether the project should be considered a packaged script.
+    ///
+    /// This detects if a project is declared as a uv package, or if it has
+    /// a scripts-related section defined.
+    pub fn is_packaged_script(&self) -> bool {
+        if let Some(is_package) = self.is_uv_package() {
+            return is_package;
+        };
+
+        if let Some(ref project) = self.project {
+            let is_script = project.gui_scripts.is_some() || project.scripts.is_some();
+            return is_script;
+        };
+
+        false
+    }
+
+    /// Returns whether the project has a `build-system` section.
+    pub fn has_build_system(&self) -> bool {
+        self.build_system.is_some()
+    }
+
+    /// Returns the value of `tool.uv.package`, if explicitly set.
+    fn is_uv_package(&self) -> Option<bool> {
```

Comment:
nit: Maybe just `uv_package`, for an `if_` function i'd have expected it returns a `bool` and not an `Option`